examples/minimal.rs

```diff
@@ -0,0 +1,384 @@
+                Err(error) => panic!("Node #{} failed to receive event: {}", index, error),
+            };
+
+            if !handle_event(index, &mut node, event, contact_tx.as_ref()) {
+                break;
+            }
+        } else {
+            // Otherwise the receiving channel is internal to node, so let it handle it.
+            if let Err(error) = node.handle_selected_operation(selected_operation) {
+                log::error!(
+                    "Node #{} failed to handle selected operation: {}",
+                    index,
+                    error
+                );
+                break;
+            }
+        }
+    }
+}
+
+// Handles the event emitted by the node.
+//
+// If `contact_tx` is `Some`, it will be used to send the contact info of the node once it becomes
+// available.
+fn handle_event(
+    index: usize,
+    node: &mut Node,
+    event: Event,
+    contact_tx: Option<&Sender<SocketAddr>>,
+) -> bool {
+    match event {
+        Event::Connected(Connected::First) => {
+            let id = node.id().expect("failed to get node name");
+            let contact_info = node
+                .our_connection_info()
+                .expect("failed to retrieve node contact info");
+            log::info!(
+                "Node #{} connected - name: {}, contact: {}",
+                index,
+                id.name(),
+                contact_info
+            );
+
+            if let Some(contact_tx) = contact_tx {
+                contact_tx
+                    .send(contact_info)
+                    .expect("failed to send contact info")
+            }
+        }
+        Event::Connected(Connected::Relocate) => {
+            let id = node.id().expect("failed to get node name");
+            log::info!("Node #{} relocated - new name: {}", index, id.name());
+        }
+        Event::Promoted => {
+            log::info!("Node #{} promoted", index);
+        }
+        Event::Demoted => {
+            log::info!("Node #{} demoted", index);
+        }
+        Event::SectionSplit(prefix) => {
+            log::info!("Node #{} section split - new prefix: {:b}", index, prefix);
+        }
+        Event::MessageReceived { content, src, dst } => log::info!(
+            "Node #{} received message - src: {:?}, dst: {:?}, content: {}",
+            index,
+            src,
+            dst,
+            HexFmt(content)
+        ),
+        Event::Consensus(payload) => log::info!(
+            "Node #{} reached consensus - payload: {}",
+            index,
+            HexFmt(payload)
+        ),
+        Event::Terminated => {
+            log::info!("Node #{} terminated", index);
+            return false;
+        }
+        Event::RestartRequired => {
+            log::info!("Node #{} requires restart", index);
+            return false;
+        }
+    }
+
+    true
+}
+
+const TARGET_SELF: &str = "minimal";
+const TARGET_ROUTING: &str = "routing";
+
+fn init_log(verbosity: u8) {
```

Comment:
Would https://github.com/rust-cli/clap-log-flag perhaps make it easier? Our CLI etc. uses this so it owudl be more consistent?

Reply:
Perhaps. But the logging logic I'm using currently is bit different. You can use `-v`, `-vv` and `-vvv` to enable logs (info, debug and trace respectively) from the routing crate only and `-vvvv` to enable logs from all crates. Doesn't seem I can achieve that with `clap-log-flag` which allows either only the current crate or all crates.

Also not sure such a simple feature justifies adding yet another dependency.